src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -137,20 +138,31 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Remark updatedRemark = personToEdit.getRemark(); // Edit command does not allow editing remarks
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
 
+        // Preserve the original orders list
+        List<Order> retainedOrders = personToEdit.getOrders();
+
         if (personToEdit instanceof Customer) {
             Information updatedInformation = editPersonDescriptor.getInformation()
                     .orElse(((Customer) personToEdit)
                     .getInformation());
-            return new Customer(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedInformation,
+            Customer editedCustomer = new Customer(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedInformation,
                     updatedRemark, updatedTags);
+
+            editedCustomer.setOrders(retainedOrders);
+            return editedCustomer;
         } else if (personToEdit instanceof Supplier) {
             Ingredients updatedIngredients = getUpdatedIngredients(editPersonDescriptor,
                     (Supplier) personToEdit, catalogue);
-            return new Supplier(updatedName, updatedPhone, updatedEmail, updatedAddress,
+            Supplier editedSupplier = new Supplier(updatedName, updatedPhone, updatedEmail, updatedAddress,
                     updatedIngredients, updatedRemark, updatedTags);
+
+            editedSupplier.setOrders(retainedOrders);
```

Comment:
think you need to update each order in retainedOrders to store the new person object as well.